docs/handshake.md

```diff
@@ -0,0 +1,1525 @@
+`HSREQ`and its own values, then sends an `HSRSP` with the effective latencies.
+
+Here is an example of an **HSv5 bidirectional transmission** between Alice and 
+Bob, where Alice is Initiator:
+
+1. Alice and Bob set the following latency values:
+
+   - Alice: `SRTO_PEERLATENCY` = 250 ms, `SRTO_RCVLATENCY` = 550 ms
+   - Bob: `SRTO_PEERLATENCY` = 500 ms, `SRTO_RCVLATENCY` = 300 ms
+
+2. Alice defines the latency field in the HSREQ message:
+```
+    hs[SRT_HS_LATENCY] = { 250, 550 }; // { Lower, Upper }
+```
+3. Bob receives it, sets his options, and responds with `HSRSP`:
+```
+    SRTO_RCVLATENCY = max(300, 250);  //<-- 250:Alice's PEERLATENCY
+    SRTO_PEERLATENCY = max(500, 550); //<-- 550:Alice's RCVLATENCY
+    hs[SRT_HS_LATENCY] = { 550, 300 };
+```
+4. Alice receives this `HSRSP` and sets:
+```
+    SRTO_RCVLATENCY = 550;
+    SRTO_PEERLATENCY = 300;
+```
+We now have the **effective latency** values:
+
+   - For transmissions from Alice to Bob: 300ms
+   - For transmissions from Bob to Alice: 550ms
+
+Here is an example of an *HSv4* exchange, which is simpler because there's only 
+one direction. We'll refer to Alice to Bob again to be consistent with the 
+Initiator/Responder roles in the HSv5 example:
+
+1. Alice sets `SRTO_LATENCY` to 250 ms
+
+2. Bob sets `SRTO_LATENCY` to 300 ms
+
+3. Alice sends `hs[SRT_HS_LATENCY] = { 250, 0 };` to Bob
+
+4. Bob does `SRTO_LATENCY = max(300, 250);`
+
+5. Bob sends `hs[SRT_HS_LATENCY] = {300, 0};` to Alice
+
+6. Alice sets `SRTO_LATENCY` to 300
+
+Note that the `SRTO_LATENCY` option in HSv5 sets both `SRTO_RCVLATENCY` and
+`SRTO_PEERLATENCY` to the same value, although when reading, `SRTO_LATENCY`
+is an alias to `SRTO_RCVLATENCY`.
+
+Why is the Sender latency updated to the effective latency for that direction?
+Because the `TLPKTDROP` mechanism, which is used by default in Live mode, may 
+cause the Sender to decide to stop retransmitting packets that are known to be 
+too late to retransmit. This latency value is one of the factors taken into 
+account to calculate the time threshold for `TLPKTDROP`.
+
+[Return to top of page](#srt-handshake)
+
+
+#### KMREQ and KMRSP
+
+`KMREQ` and `KMRSP` contain the KMX (key material exchange) message used for
+encryption. The most important part of this message is the
+AES-wrapped key (see the [Encryption documentation](encryption.md) for
+details). If the encryption process on the Responder side was successful,
+the response contains the same message for confirmation. Otherwise it's
+one single 32-bit value that contains the value of `SRT_KMSTATE` type,
+as an error status.
+
+Note that when the encryption settings are different at each end, then
+the connection is still allowed, but with the following restrictions:
+
+- If the Initiator declares encryption, but the Responder does not, then
+the Responder responds with `SRT_KM_S_NOSECRET` status. This means
+that the Responder will not be 
```

Comment:
To avoid confusion here, I'd say it explicitly that it's a string `that a Caller can pass to a Listener (it's actually passed from an Initiator to a Responder in general, but in Rendezvous connections this feature doesn't make sense)`.